examples/nginx.conf

```
@@ -12,67 +12,95 @@ http {
   js_path "/usr/lib/nginx/njs_modules/";
   js_fetch_trusted_certificate /etc/ssl/certs/ISRG_Root_X1.pem;
 
+  # Read the .js file into the `acme` namespace.
   js_import acme from acme.js;
 
-  # One `resolver` directive must be defined.
+  # IMPORTANT: One `resolver` directive *must* be defined.
   resolver 127.0.0.11 ipv6=off; # docker-compose
   # resolver 1.1.1.1 1.0.0.1 [2606:4700:4700::1111] [2606:4700:4700::1001] valid=300s; # Cloudflare
   # resolver 8.8.8.8 8.8.4.4; # Google
   # resolver 172.16.0.23; # AWS EC2 Classic
   # resolver 169.254.169.253; # AWS VPC
   resolver_timeout 5s;
 
-  ## advanced use-case for njs-acme to optionally use shared dict to cache cert/key pairs
-  # then provide the same zone name in $njs_acme_shared_dict_zone_name
-  # zone size should be enough to store all certs and keys; 1MB should be enough to store 100 certs/keys
+  ##
+  # `njs-acme` can use a shared dict to cache cert/key pairs to avoid
+  # filesystem calls on TLS handshake.  If you want to use a shared zone name
+  # that is not `acme`, then ensure the variable $njs_acme_shared_dict_zone_name
+  # also contains the desired name.  The zone size should beenough to store all
+  # certs and keys. 1MB should be enough to store 100 certs/keys.
   js_shared_dict_zone zone=acme:1m;
 
+  ##
+  # Set up a specific server to listen on the localhost interface for
+  # js_periodic to trigger acme.clientAutoMode. The default URL is
+  # http://localhost:10999/acme/auto but can be overriden via the
+  # NJS_ACME_CLIENT_AUTO_MODE_URL environment variable.
   server {
-    listen 0.0.0.0:8000; # testing with 8000 should be 80 in prod, pebble usees httpPort in dev/pebble/config.json
+    listen 127.0.0.1:10999;
+    # Check for certificate expiration every minute
+    location @acmePeriodicAuto {
+      js_periodic acme.periodicAuto interval=1m;
```

dumb question: can we just call the function instead of making an HTTP request that calls a function?

```suggestion
      js_periodic acme.clientAutoMode interval=1m;
```

---

Great suggestion and this is what's doing now. Thanks @ryepup !